rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java

```diff
@@ -1025,4 +1045,14 @@ public boolean hasPendingActions() {
 
         return participants;
     }
+
+    public static final class LRAStatusReason {
```

Comment:
Is the correct name for the `enum "LRAStatusReason" when it seems it gathers only failure states? Is there a expectation that some additional non-error statuses will be added?

Reply:
The set of reasons for an LRA to be in a given state is finite so use an enum to model it. If you really think this structure needs to be extensible (which we should strive to avoid) then introduce a value that can be further embellished (such as `Other` or `Unknown`) by using a secondary data type.

Note that I did not think about whether you have captured the full set of reasons for each state transition in our model (there are reasons why a state is left and reasons why a state is entered so the values need to capture both reasons). If you can wait, I will investigate this tonight (I'm on PTO this week).